### PR TITLE
fix: update allowances

### PR DESCRIPTION
**Pull request by marktoda**

### CODE REVIEW

File: components/Dashboard.tsx [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
 
   if (!spendersAndTokens) return <div>Loading...</div>
 
-  return <div>{spendersAndTokens?.map(([spender, tokens]) => (
+  return <div>{Object.keys(spendersAndTokens)?.map((spender) => (
     <div key={spender}>
       Spender: {spender} <br />
-      {tokens.map(token => <div key={token}>Token: {token}</div>)}
+      {Object.entries(spendersAndTokens[spender]).map(([token, { amount, expiration }]) => (
+        <div>
```
Comment (tinaszheng) [inline components/Dashboard.tsx]:
```suggestion
        <div key={token}>
```

File: components/Dashboard.tsx [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
+          <div key={token}>Token: {token}</div>
```
Comment (tinaszheng) [inline components/Dashboard.tsx]: u can take the keys out of these child components

File: permit2.ts [hunk 3]
```diff
@@ -1,52 +1,74 @@
...
+  )
 
-  const results = await multicallProvider.all(calls);
+  const results = await multicallProvider.all(calls)
 
   return results.reduce((allowances, [amount, expiration], i) => {
-    const { token, spender } = tokenSpenders[i];
-    if (!allowances[token]) {
-      allowances[token] = {};
+    const { token, spender } = tokenSpenders[i]
```
Comment (tinaszheng) [inline permit2.ts:57]: dis kinda messy

can it be a for loop instead xD

**Review by tinaszheng — APPROVED** [review]

File: permit2.ts (hunk 3 above)
Comment (marktoda) [reply permit2.ts:57]: waa you dont like functional style?

Comment (tinaszheng) [reply permit2.ts:57]: bruhhh

**Pull request merged**